### PR TITLE
feat(map_based_prediction): enable lane change detection from unconnected lane

### PR DESCRIPTION
**Pull request by YoshiRi**
### TL;DR

- Currently map based prediction can not generate lane change detection paths from unconnected lane(cf. see the figure below)
- This PR enable to predict lane change to unconnected adjacent lane
  - Paths changing from the forward lane to the opposite lane are not predicted now
  - Therefore, cases in which the driver once moves out into the opposite lane to avoid on-street parking, etc., are not included.

image

### Background

Current prediction path generation for on-lane object is summarized as follows:

1. Extract left/right/center lanelets and paths
2. Calculate lane change probability
3. Generate actual paths for possible paths

in this part.1 uses `left()` function, which can only get connected left lane. Also, `adjacentLeft` can only get same direction lane, so that we need to extract opposite lane manually. (cf. see https://github.com/fzi-forschungszentrum-informatik/Lanelet2/issues/147)

### Changes

- update searching lanelet path candidate search in `getPredictedReferencePath`
- changed current lanelet estimation in `getCurrentLanelets`

####  lanelets candidate searching

In `getPredictedReferencePath` function, we updated the predicted path candidate extraction so that it can predict the path from opposite lane to current lane cut-in

- get left/right lanelet to create a candidate path
- if left/right lanelet is empty, try to get adjacentLeft/adjacentRight lanelet
- if all above are empty, try to get opposite lane
  - if object is trying to change lane to opposite lane, do not predict that

#### get current lanet function update

- allow current lanelet search function to search opposite lanelet only if there are no possible lanelets in forward lanelets

### Links

- [TIER IV internal tickets](https://tier4.atlassian.net/browse/T4PB-26205)

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by takayuki5168** [comment]
@YoshiRi 
It seems it does not work well with kashiwanoha and odaiba map.
image
image

**Comment by YoshiRi** [comment]
@takayuki5168 
Thanks for your check. Did it checked by the scenario simulator or psim? May be I can try with psim by myself.

**Comment by takayuki5168** [comment]
@YoshiRi psim

**Comment by YoshiRi** [comment]
This trouble caused by missed current lanelet estimation. This is solved in https://github.com/autowarefoundation/autoware.universe/pull/3007/commits/bb28db7afe772b28650c7e4952569567adc83be9.

**Comment by takayuki5168** [comment]
@YoshiRi 
When spawning the car with the `k` button on the planning simulator as follows, the predicted path is wired.
- The path on the opposite lanelet appears which is not expected.
    - and its path end is unstable
    - and it does not have confidence.
image

Also on the terminal, the warning `[WARN 1680541783.720588538] [BigVehicleTracker]: Cannot update (measureWithPose():343)` is always output when the obstacle is running.

**Comment by Mingyu1991** [comment]
@takayuki5168 @miursh @tkimura4 @purewater0901 
this is just a friendly reminder that this PR is not active for more than 1 week.

**Comment by takayuki5168** [comment]
@YoshiRi This feature was merged into main with another PR. Could you close the PR if it's okay?
